addons/l10n_ar/tests/common.py

```diff
@@ -0,0 +1,1164 @@
+                        invoice_line_form.price_unit = line.get('price_unit', 100)
+            invoice_form.invoice_date = invoice_form.date
+        invoice = invoice_form.save()
+        return invoice
+
+    def _create_invoice_product(self, data=None):
+        data = data or {}
+        return self._create_invoice(data)
+
+    def _create_invoice_service(self, data=None):
+        data = data or {}
+        newlines = []
+        for line in data.get('lines', [{}]):
+            line.update({'product': self.service_iva_27})
+            newlines.append(line)
+        data.update({'lines': newlines})
+        return self._create_invoice(data)
+
+    def _create_invoice_product_service(self, data=None):
+        data = data or {}
+        newlines = []
+        for line in data.get('lines', [{}]):
+            line.update({'product': self.product_iva_21})
+            newlines.append(line)
+        data.update({'lines': newlines + [{'product': self.service_iva_27}]})
+        return self._create_invoice(data)
+
+    def _create_credit_note(self, invoice, data=None):
+        data = data or {}
+        refund_wizard = self.env['account.move.reversal'].with_context({'active_ids': [invoice.id], 'active_model': 'account.move'}).create({
+            'reason': data.get('reason', 'Mercadería defectuosa'),
+            'refund_method': data.get('refund_method', 'refund')})
+
+        forced_document_type = data.get('document_type')
+        if forced_document_type:
+            refund_wizard.l10n_latam_document_type_id = forced_document_type.id
+
+        res = refund_wizard.reverse_moves()
+        refund = self.env['account.move'].browse(res['res_id'])
+        return refund
+
+    def _create_debit_note(self, invoice, data=None):
+        data = data or {}
+        debit_note_wizard = self.env['account.debit.note'].with_context(
+            {'active_ids': [invoice.id], 'active_model': 'account.move', 'default_copy_lines': True}).create({
+                'reason': data.get('reason', 'Mercadería defectuosa')})
+        res = debit_note_wizard.create_debit()
+        debit_note = self.env['account.move'].browse(res['res_id'])
+        return debit_note
+
+    def _search_tax(self, tax_type):
+        res = self.env['account.tax'].with_context(active_test=False).search([
+            ('type_tax_use', '=', 'sale'),
+            ('company_id', '=', self.env.company.id),
+            ('tax_group_id', '=', self.env.ref('l10n_ar.tax_group_' + tax_type).id)], limit=1)
+        self.assertTrue(res, '%s Tax was not found' % (tax_type))
+        return res
+
+    def _search_fp(self, name):
+        return self.env['account.fiscal.position'].search([('company_id', '=', self.env.company.id), ('name', '=', name)])
+
+    def _post(self, invoice):
+        invoice.action_post()
+
+    def _validate_and_review(self, invoice, expected_result=None, error_msg=None):
+        """ Validate an invoice and review that the invoice has been properly posted. """
```

Comment:
This method is weird.  It only checks if the invoice is posted, but how can you have an expected_result with respect to that?